src/main/java/study/StringAddCalculator.java

```diff
@@ -0,0 +1,48 @@
+package study;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+	public static int splitAndSum(String input) throws RuntimeException {
+		if (isEmpty(input))
+			return 0;
+
+		if (input.contains("-"))
+			throw new RuntimeException();
+
+		Matcher matcher = patternMatcher(input);
+
+		String[] numbers;
+		if (matcher.find()) {
+			String customDelimiter = matcher.group(1);
+			String text = matcher.group(2);
+			numbers = split(text, customDelimiter);
+			return sum(numbers);
+		}
```

Comment:
else 예약어 없이 잘 구현해주셨네요!
다만, if와 하단 부분의 로직이 겹치는 부분이 있는 것 같아요!

```java
private static final String DEFAULT_DELIMITER = ",|:";
```
기본 구분자를 선언한 뒤,

```java
private static String findDelimiter(String text) {
    Matcher m = PATTERN.matcher(text); 
    if (m.find()) { 
        return DEFAULT_DELIMITERS + '|' + m.group(1); 
    } 
    return DEFAULT_DELIMITERS; 
}
```

구분자를 결정하는 메서드를 선언하는 방법으로 리팩토링 할 수 있을것 같아요!

```suggestion
		String delimiter = findDelimiter(text);
	        String[] numbers = split(text, delimiter);
	        return sum(numbers);
```